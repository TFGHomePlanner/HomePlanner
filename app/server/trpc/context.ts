import * as trpc from "@trpc/server";
import { prisma } from "../../common/prisma";
import * as trpcExpress from "@trpc/server/adapters/express";

export async function createContext(
  ctx: trpcExpress.CreateExpressContextOptions
) {
  const { req, res } = ctx;
  return {
    req,
    res,
    prisma,
  };
}

export type Context = trpc.inferAsyncReturnType<typeof createContext>;
