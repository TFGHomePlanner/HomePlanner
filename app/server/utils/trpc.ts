import { createTRPCReact } from "@trpc/react-query";
import type { AppRouter } from "../trpc/router/_app";
import { httpBatchLink } from "@trpc/client";

export const trpc = createTRPCReact<AppRouter>({});