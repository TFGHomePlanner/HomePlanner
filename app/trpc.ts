import { createTRPCReact } from "@trpc/react-query";
import type { AppRouter } from "./server/trpc/router/_app";


export const trpc = createTRPCReact<AppRouter>();
