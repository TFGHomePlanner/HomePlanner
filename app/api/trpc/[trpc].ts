import * as trpcNext from "@trpc/server/adapters/next";
import { appRouter } from "../../server/trpc/router/_app";
import { createContext } from "../../server/trpc/context";

export default trpcNext.createNextApiHandler({
	router: appRouter,
	createContext,
});