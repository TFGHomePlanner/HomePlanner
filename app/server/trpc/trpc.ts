import { initTRPC } from "@trpc/server";
import { Context } from "./context";
/**
 * Inicializaci√≥n de la instancia tRPC en backend
 */
const t = initTRPC.context<Context>().create();

/**
 * Exporta el router reutilizable y helpers de ayuda
 * que pueden ser utilizados en todo el router.
 */
export const router = t.router;
export const publicProcedure = t.procedure;
