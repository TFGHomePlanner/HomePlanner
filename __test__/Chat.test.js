import React from "react";
import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import trpc from "../app/server/utils/trpc";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { httpBatchLink } from "@trpc/client";
import { test, expect } from "@jest/globals";
import LoginScreen from "../app/screens/login";



test("renders correctly", () => {
  const tree = renderer.create(
    <LoginScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
