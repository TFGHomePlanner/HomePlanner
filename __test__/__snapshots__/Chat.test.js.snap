// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#F8F3ED",
        },
      ],
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": -40,
        },
        {
          "marginBottom": 32,
        },
        {
          "fontSize": 24,
          "lineHeight": 32,
        },
        {
          "fontWeight": "600",
        },
        {
          "color": "#212529",
        },
      ]
    }
  >
    homeplanner
  </Text>
  <View
    style={
      [
        {
          "width": 288,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="email"
      placeholderTextColor="#929193"
      style={
        [
          {
            "marginBottom": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
          {
            "borderBottomColor": "#2125292c",
            "borderLeftColor": "#2125292c",
            "borderRightColor": "#2125292c",
            "borderTopColor": "#2125292c",
          },
          {
            "backgroundColor": "#E5E5E5",
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          {
            "color": "#212529",
          },
        ]
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="contraseña"
      placeholderTextColor="#929193"
      style={
        [
          {
            "marginBottom": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
          {
            "borderBottomColor": "#2125292c",
            "borderLeftColor": "#2125292c",
            "borderRightColor": "#2125292c",
            "borderTopColor": "#2125292c",
          },
          {
            "backgroundColor": "#E5E5E5",
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          {
            "color": "#212529",
          },
        ]
      }
      value=""
    />
    <Text
      style={
        [
          {
            "marginBottom": 8,
          },
          {
            "alignSelf": "flex-end",
          },
          {
            "fontSize": 12,
            "lineHeight": 16,
          },
          {
            "fontWeight": "600",
          },
          {
            "color": "#1E88E5",
          },
        ]
      }
    >
      ¿Has olvidado tu contraseña?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "marginBottom": 16,
            "marginTop": 16,
          },
          {
            "width": "100%",
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
          {
            "backgroundColor": "#1E88E5",
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "center",
            },
            {
              "fontWeight": "500",
            },
            {
              "color": "#F8F3ED",
            },
          ]
        }
      >
        Iniciar sesión
      </Text>
    </View>
    <Text
      style={
        [
          {
            "marginTop": 8,
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "#929193",
          },
        ]
      }
    >
      ¿Todavía no tienes cuenta?
       
      <Text
        accessibilityRole="link"
        href="/Register"
        onPress={[Function]}
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "color": "#1E88E5",
              },
            ]
          }
        >
          Únete
        </Text>
      </Text>
    </Text>
  </View>
</View>
`;
